### PR TITLE
DBZ-1356 Metrics for schema history

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-1356

### CODE REVIEW

File: debezium-core/src/main/java/io/debezium/metrics/Metrics.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+ * Base for metrics implementations.
+ *
+ * @author Jiri Pechanec
+ */
+@ThreadSafe
+public abstract class Metrics {
+
+    private final ObjectName name;
+    
+    protected <T extends CdcSourceTaskContext> Metrics(T taskContext, String contextName) {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/metrics/Metrics.java]: `T` seems unused?

File: debezium-core/src/main/java/io/debezium/metrics/Metrics.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+    protected Metrics(CommonConnectorConfig connectorConfig, String contextName) {
+        this.name = metricName(connectorConfig.getConnectorType(), connectorConfig.getLogicalName(), contextName);
+    }
+
+    /**
+     * Registers a metrics MBean into the platform MBean server.
+     * The method is intentionally synchronized to prevent preemption between registration and unregistration.
+     */
+    public synchronized <T extends CdcSourceTaskContext> void register(Logger logger) {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/metrics/Metrics.java]: `T` seems unused?

File: debezium-core/src/main/java/io/debezium/pipeline/metrics/Metrics.java [hunk 3]
```diff
@@ -30,62 +24,26 @@
  * @author Randall Hauch, Jiri Pechanec
  */
 @ThreadSafe
-public abstract class Metrics implements DataChangeEventListener, ChangeEventSourceMetricsMXBean {
+public abstract class Metrics extends io.debezium.metrics.Metrics implements DataChangeEventListener, ChangeEventSourceMetricsMXBean {
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/pipeline/metrics/Metrics.java]: Should we rename this into `PipelineMetrics` then?

File: debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseSchema.java [hunk 4]
```diff
@@ -34,7 +35,7 @@ protected HistorizedRelationalDatabaseSchema(HistorizedRelationalDatabaseConnect
         super(config, topicSelector, tableFilter, columnFilter, schemaBuilder, tableIdCaseInsensitive);
 
         this.databaseHistory = config.getDatabaseHistory();
-        this.databaseHistory.start();
+        this.databaseHistory.start(new DatabaseHistoryMetrics(config));
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/HistorizedRelationalDatabaseSchema.java]: Can this be passed to to `DatabaseHistory#configure()` instead?

File: debezium-core/src/main/java/io/debezium/relational/history/DatabaseHistoryMetrics.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+ * Implementation of {@link DatabaseSchema} metrics.
+ *
+ * @author Jiri Pechanec
+ *
+ */
+public class DatabaseHistoryMetrics extends Metrics implements DatabaseHistoryListener, DatabaseHistoryMXBean {
+
+    private static final String CONTEXT_NAME = "schema-history";
+
+    private static final int MILLISECONDS_BETWEEN_LOG_MESSAGES = 2_000;
```
Comment (gunnarmorling) [inline debezium-core/src/main/java/io/debezium/relational/history/DatabaseHistoryMetrics.java]: Should be a `Duration`.

**Comment by jpechane** [comment]
@gunnarmorling All requests incorporated

**Comment by gunnarmorling** [comment]
Squashed and applied. Thx, @jpechane!

**Pull request merged**